app/workers/historical_data_deletion_worker.rb

```diff
@@ -37,9 +37,23 @@ def delete_and_log(model)
     logger.info(message)
   end
 
-  def empty_subscriber_lists
-    SubscriberList.left_outer_joins(:subscriptions)
-                  .where("subscriber_lists.created_at < ?", empty_list_retention_period)
-                  .where("subscriptions.id": nil)
+  def historic_subscriber_lists
+    subscriptions_exist = Subscription.where(
+      "subscriber_lists.id = subscriptions.subscriber_list_id",
+    ).arel.exists
+
+    SubscriberList
+      .where("created_at < ?", empty_list_retention_period)
+      .where.not(subscriptions_exist)
+  end
+
+  def historic_subscribers
+    subscriptions_exist = Subscription.where(
+      "subscribers.id = subscriptions.subscriber_id",
+    ).arel.exists
```

Comment:
It'd be useful if the commit had metrics on the performance increase, saying "much quicker" leaves it rather ambiguous.

Reply:
This may be in reference to [a comment I made](https://github.com/alphagov/email-alert-api/pull/1462#discussion_r518882513). I suspect it was actually the `group` that was causing the original query there to be so slow, but I only have the evidence of it not completing in a number of minutes.

I'm not sure it makes much difference for the queries here. That said, I do find the proposed change easier to read, as it's more consistent and matches the intuitive idea of checking each subscriber individually.

Reply:
I've generally found joins to be quite slow in general versus multiple query approaches but I'll pop the exact benchmarks below.

With the `join` approach when finding historic subscriber lists the query ran in around 2700ms.

With the sub query approach it ran in around 167ms. 

Only 16 times quicker! 😆 

Have also added this info to the commit.